pymesync.py

```diff
@@ -83,6 +83,57 @@ def get_times(self, **queries):
             # Request Error
             return e
 
+    def get_projects(self, **kwargs):
+        """
+        get_times(kwargs)
+
+        Returns JSON projects objects filtered by supplied parameters
+        Optional parameters:
+        slug='<slug>'
+        include_deleted=<boolean>
+        revisions=<boolean>
+
+        Does not accept a slug and include_deleted, but does accept any other
+        combination.
+        """
+        query_string = ""
+        query_list = []
+        if kwargs:
+            # The following combination is not allowed
+            if 'slug' in kwargs.keys() and 'include_deleted' in kwargs.keys():
+                return "Error: invalid combination of slug and include_deleted"
+            # slug goes first, then delete it so it doesn't show up after the ?
+            elif 'slug' in kwargs.keys():
+                query_string = "/{}".format(kwargs['slug'])
+                del(kwargs['slug'])
+
+            # Convert True and False booleans to TimeSync compatible strings
+            for k, v in sorted(kwargs.items(), key=operator.itemgetter(0)):
+                if v is True:
+                    kwargs[k] = 'true'
+                elif v is False:
+                    kwargs[k] = 'false'
+                query_list.append("{0}={1}".format(k, kwargs[k]))
+
+            # Check for items in query_list after slug was removed, create
+            # query string
+            if query_list:
+                query_string += "?{}".format("&".join(query_list))
+
+        # Construct query url - query_string is empty if no kwargs
+        url = "{0}/{1}/projects{2}".format(self.baseurl,
+                                           self._api_version(),
+                                           query_string)
+
+        # Attempt to GET projects
+        try:
+            # Success!
+            response = requests.get(url)
+            return response
```

Comment:
@Kennric I'm working on this, but it is proving harder to test than I thought since I'm mocking `requests.get(url)`. It is creating failures because I can't create a python object from a mocked return.